### PR TITLE
Fix unintended field replacement on sources

### PR DESCRIPTION
**Pull request by aelesbao**
Fields that were referenced on the `MultiSourceSelect.querySpec` and
on a `source` where being replaced on both places by relation columns.
It is expected that the replacement occurrs only on the first.

### CODE REVIEW

**Comment by aelesbao** [comment]
@smu could you please review?

File: CHANGES.txt [hunk 1]
```diff
@@ -5,6 +5,8 @@ Changes for Crate
 Unreleased
 ==========
 
+ - Fixed an issue when ordering by a function on a cross join query.
```
Comment (seut) [inline CHANGES.txt]: is this only related to cross joins (what about inner joins)?

**Comment by seut** [comment]
lgtm
